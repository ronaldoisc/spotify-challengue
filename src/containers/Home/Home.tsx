import React, { useEffect } from 'react';
import styled from 'styled-components';

import { useDispatch, RootStateOrAny, useSelector } from 'react-redux';
import { Navbar } from '../../components/Navbar';
import { startLoadingNewReleases } from '../../redux/modules/newReleases';
import { Releases } from '../Releases/Releases';
import { getCurrentTime } from '../../helper/getCurrentTime';


const Time = styled.h1`
margin:0.7rem;
color:white;
`;

export const Home = () => {
  const time = getCurrentTime();
  const dispatch = useDispatch();

  const { ui, newReleases } = useSelector((state: RootStateOrAny) => state);
  const { releases } = newReleases;
  const { isLoading } = ui;
  
  useEffect(() => {
    dispatch(startLoadingNewReleases());
  }, [dispatch])

 
  return (
    <>
      {
        isLoading ?
          <h1>Cargando....</h1>
          :
          <div>
            <Navbar />
            <Time>{time}!</Time>
            <Releases releases={releases} />
          </div>
      }
    </>
  )
}
