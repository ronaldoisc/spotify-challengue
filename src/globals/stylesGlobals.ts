import { createGlobalStyle } from 'styled-components';


export const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
}
export const device = {
    mobileS: `(min-width: ${size.mobileS})`,
    mobileM: `(min-width: ${size.mobileM})`,
    mobileL: `(min-width: ${size.mobileL})`,
    tablet: `(min-width: ${size.tablet})`,
    laptop: `(min-width: ${size.laptop})`,
    laptopL: `(min-width: ${size.laptopL})`,
    desktop: `(min-width: ${size.desktop})`,
    desktopL: `(min-width: ${size.desktop})`
};

export const GlobalCSS = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    background-color: rgb(18,18,18);
    color:white;
    font-family: 'Poppins', sans-serif;
  }
`;
