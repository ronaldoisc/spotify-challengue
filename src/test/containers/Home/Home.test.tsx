import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';

import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { Home } from '../../../containers/Home/Home';
import { ThemeProvider } from "styled-components";
import { theme } from '../../../globals/stylesGlobals';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    newReleases: {
        releases: []
    },
    ui: {
        isLoading: false
    }
};
let store = mockStore(initialState);
describe('test in the container folder', () => {

    test('should show the Home component correctly', async () => {

        const wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <Home />
                </ThemeProvider>
            </Provider>)

        expect(wrapper).toMatchSnapshot();

    });


})