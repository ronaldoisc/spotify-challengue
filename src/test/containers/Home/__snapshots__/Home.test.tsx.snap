// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in the container folder should show the Home component correctly 1`] = `
<div>
  <div>
    <header
      className="sc-bdvvtL dTRxGY"
    >
      <img
        alt="logo"
        height={40}
        src="banner-ico.png"
        width={40}
      />
      <h1
        className="sc-gsDKAQ dtdhG"
      >
        spotify
      </h1>
    </header>
  </div>
  <h1
    className="sc-iCfMLu dubHju"
  >
    Good Afternoon
    !
  </h1>
  Array [
    <h1
      className="sc-jRQBWg gXLsvu animate__animated animate__fadeIn"
    >
      New Releases
    </h1>,
    <section
      className="sc-hKwDye obivw"
    />,
  ]
</div>
`;
